lua/nvim-treesitter/query_predicates.lua

```diff
@@ -85,5 +86,34 @@ query.add_predicate('has-type?', function(match, pattern, bufnr, pred)
   return vim.tbl_contains(types, node:type())
 end)
 
+
+-- Inject a language based on the pattern from `config.injections.pattern`.
+-- Usage: (#inject! @language [default])
+query.add_directive('inject!', function(match, pattern, bufnr, pred, metadata)
+  local match_id = pred[2]
+  local default = pred[3] or ''
+  local node = match[match_id]
+  local text = query.get_node_text(node, bufnr)
+  local configs = parsers.get_parser_configs()
+  local language = default
+  for lang in pairs(configs) do
+    if text == lang then
```

Comment:
why do a reverse look up? Going through all languages and all available regexes? Wouldn't one table with ["py"] = "python" faster and less distributed to configure? Normalization like lowercase could be done before.

Reply:
Maybe take an already existing table: https://github.com/tjdevries/plenary.nvim/blob/9bd408ba679d1511e269613af840c5019de59024/data/plenary/filetypes/base.lua#L1

Reply:
i agree, a table will speed up the directive impl

Reply:
Agreed on the table for lookup here too

Reply:
👍

Reply:
A CI jobs could also help generate parts of the table via their ftdetect table. I imagine the mapping extension to language could be handy for example in Markdown.